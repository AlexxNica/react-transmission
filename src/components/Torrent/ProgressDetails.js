import React from 'react';

function ProgressDetails({ torrent }) {
  return (
    <p>Downloading ... ↓ {torrent.rateDownload} ↑ {torrent.rateUpload}</p>
  );
}

export default ProgressDetails;
