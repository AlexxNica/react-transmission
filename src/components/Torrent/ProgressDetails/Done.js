import React from 'react';
import { FormattedMessage, defineMessages, injectIntl } from 'react-intl';

import { size as formatSize, ratioString } from 'util/formatters';

const messages = defineMessages({
  uploadStats: {
    id: 'torrent.progress.done.uploadstats',
    defaultMessage: `uploaded {uploaded} (Ratio: {ratio})`,
  },
});

const PartialDone = injectIntl(({ torrent, intl }) =>
  <FormattedMessage
    id='torrent.progress.done.partial'
    defaultMessage={`{remaining} of {total} ({completed}), {stats}`}
    values={{
      remaining: formatSize(torrent.sizeWhenDone),
      total: formatSize(torrent.totalSize),
      completed: intl.formatNumber(torrent.percentDone, {style: 'percent'}),
      stats: intl.formatMessage(messages.uploadStats, {
        uploaded: formatSize(torrent.uploadedEver),
        ratio: ratioString(torrent.uploadRatio),
      }),
    }}
  />
);

const DoneDone = injectIntl(({ torrent, intl }) =>
  <FormattedMessage
    id='torrent.progress.done.done'
    defaultMessage={`{total}, {stats}`}
    values={{
      total: formatSize(torrent.totalSize),
      stats: intl.formatMessage(messages.uploadStats, {
        uploaded: formatSize(torrent.uploadedEver),
        ratio: ratioString(torrent.uploadRatio),
      }),
    }}
  />
);

function ProgressDone({ torrent }) {
  if (torrent.totalSize === torrent.sizeWhenDone) {
    return <DoneDone torrent={torrent} />;
  }

  return <PartialDone torrent={torrent} />;
}

export default ProgressDone;
