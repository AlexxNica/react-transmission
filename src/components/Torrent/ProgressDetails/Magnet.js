import React from 'react';
import { FormattedMessage, injectIntl, defineMessages } from 'react-intl';

const messages = defineMessages({
  needs: {
    id: 'torrent.progress.magnet.needs',
    defaultMessage: 'needs',
  },
  retrieving: {
    id: 'torrent.progress.magnet.retrieving',
    defaultMessage: 'retrieving',
  },
});

function ProgressMagnet({ torrent, intl }) {
  const metadataStatus = torrent.isStopped ? messages.needs : messages.retrieving;

  return (
    <FormattedMessage
      id='torrent.progress.magnet'
      defaultMessage={`Magnetized transfer - {metadataStatus} metadata ({completed})`}
      values={{
        metadataStatus: intl.formatMessage(metadataStatus),
        completed: intl.formatNumber(torrent.metadataPercentComplete, {style: 'percent'}),
      }}
    />
  );
}

export default injectIntl(ProgressMagnet);
