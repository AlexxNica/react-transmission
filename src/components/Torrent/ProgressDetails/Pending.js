import React from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';

import { size as formatSize } from 'util/formatters';

function ProgressPending({ torrent, intl }) {
  const { sizeWhenDone, leftUntilDone, percentDone } = torrent;

  return (
    <FormattedMessage
      id='torrent.progress.pending'
      defaultMessage={`{remaining} of {total} ({completed})`}
      values={{
        remaining: formatSize(sizeWhenDone - leftUntilDone),
        total: formatSize(sizeWhenDone),
        completed: intl.formatNumber(percentDone, {style: 'percent'}),
      }}
    />
  );
}

export default injectIntl(ProgressPending);
