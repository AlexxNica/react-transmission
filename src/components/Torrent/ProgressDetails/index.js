import React from 'react';

import Magnet from './Magnet';
import Done from './Done';
import Pending from './Pending';

export default function({ torrent }) {
  if (torrent.needsMetaData) {
    return <Magnet torrent={torrent} />;
  }

  // TODO: Append ETA
  if (torrent.isDone || torrent.isSeeding) {
    return <Done torrent={torrent} />;
  }

  return <Pending torrent={torrent} />;
}
