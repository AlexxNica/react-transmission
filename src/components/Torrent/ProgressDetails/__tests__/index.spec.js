import React from 'react';
import createComponentWithIntl from 'test/util/createComponentWithIntl';

import ProgressDetails from '../';

const torrent = {
  metadataPercentComplete: 0.75,
  sizeWhenDone: 1E5,
  leftUntilDone: 1E4,
  percentDone: 0.9,
  totalSize: 1E5,
  uploadedEver: 1E5,
  uploadRatio: 2,
};

test('ProgressDetails magnet', () => {
  const component = createComponentWithIntl(
    <ProgressDetails torrent={{...torrent, needsMetaData: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('ProgressDetails done', () => {
  const component = createComponentWithIntl(
    <ProgressDetails torrent={{...torrent, isDone: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('ProgressDetails seeding', () => {
  const component = createComponentWithIntl(
    <ProgressDetails torrent={{...torrent, isSeeding: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('ProgressDetails pending', () => {
  const component = createComponentWithIntl(
    <ProgressDetails torrent={{...torrent}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
