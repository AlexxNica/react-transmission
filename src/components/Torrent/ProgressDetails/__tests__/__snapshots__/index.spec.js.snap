exports[`test ProgressDetails done 1`] = `
<span>
  100.0 kB, uploaded 100.0 kB (Ratio: 2.00)
</span>
`;

exports[`test ProgressDetails magnet 1`] = `
<span>
  Magnetized transfer - retrieving metadata (75%)
</span>
`;

exports[`test ProgressDetails pending 1`] = `
<span>
  90.0 kB of 100.0 kB (90%)
</span>
`;

exports[`test ProgressDetails seeding 1`] = `
<span>
  100.0 kB, uploaded 100.0 kB (Ratio: 2.00)
</span>
`;
